DAT 3732 Infobox builder header section editing

RafLeszczynski opened the pull request:
https://wikia-inc.atlassian.net/browse/DAT-3732

Should be merged together with:
https://github.com/Wikia/mercury/pull/2055

rybmat committed:
support for collapse attrib in group node while converting xml->json

rybmat committed:
support for collapse=open in group node while converting from json to xml

rybmat committed:
cleanup

rybmat committed:
cleanup

rybmat committed:
extracting logic creating group dom to one method

Commit:
DAT-3732, move collpsible group chevron styles to mixin and use it in preview

rybmat committed:
checking if group doesn't have collapse attr other than 'open'

rybmat committed:
refactor of handling collapsible sections in json->xml translation

rybmat committed:
Merge branch 'DAT-3732' of github.com:Wikia/app into DAT-3732

rybmat committed:
messages in markup support test

rybmat committed:
comments, casting to bool

Commit:
Merge branch 'dev' into DAT-3732

dianafa commented on extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:
```diff
@@ -16,7 +16,36 @@ class NodeGroup extends Node {
+		}
+
+		foreach ( $data as $child) {
+			if ( $child->type == 'section-header') {
```
strange spacing around brackets here and above

dianafa commented on extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:
```diff
@@ -16,7 +16,36 @@ class NodeGroup extends Node {
+
+	protected function hasValidAttributes() {
+		foreach ( $this->xmlNode->attributes() as $attr => $val ) {
+			if ( $attr == 'collapse' && $val != 'open') {
```
why not `===` and `!==` ?

dianafa commented on extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:
```diff
@@ -210,9 +219,13 @@ protected function getFormattedMarkup( $xml, $formatted ) {
 					// and initialize a new group
-					$currentGroupDom = $this->createGroupDom();
-					$currentGroupDom->appendChild( $currentGroupDom->ownerDocument->importNode( $childNodeDom, true ) );
+					$currentGroupDom = $this->createGroupDom($childNodeDom, $collapsible);
```
this code is the same as in lines 203-209. Maybe we can move it to a separate function?

sebastianmarzjan replied:
This is intentional here - since childNodeDom is mutated, doing it in separate function would decrease the readabiilty of this code

dianafa commented:
@rybmat in MW app code we put spaces between `(` `)` and other char, that is instead of `createGroupDom($childNodeDom, $collapsible)` should be `createGroupDom( $childNodeDom, $collapsible )`

Commit:
DAT-3732, fix whitespace issues

dianafa committed:
add space

dianafa commented:
LGTM

dianafa commented:
tests are failing! :(

Commit:
Fix tests failing after introduction of strict comparison ( '===' instead of '==' ).
In these cases, automatic type cast does not happen and one needs to cast the SimpleXML resource
to string explicitly.

Commit:
DAT-3732, support empty headers

Commit:
DAT-3732, fix issue with empty header height

Commit:
DAT-3732, add test for empty header
